File: ExpenseControl.Core/Services/PersonService.cs
using ExpenseControl.Core.Entities;
using ExpenseControl.Core.Interfaces;

namespace ExpenseControl.Core.Services;


/// <summary>
///  Serviço responsável pelas regras de negócio da entidade Person.
/// Realiza as validações antes de delegar a persistência ao repositório.
/// </summary>
public class PersonService : IPersonService
{
    private readonly IPersonRepository _personRepository;
    
    public  PersonService(IPersonRepository personRepository)
    {
        _personRepository = personRepository;
    }

    /// <summary>
    /// Cria uma nova pessoa após validar os campos obrigatórios.
    /// </summary>
    /// <param name="person">A entidade pessoa a ser criada.</param>
    /// <returns>A pessoa criada com o identificador único atribuído pelo banco de dados.</returns>>
    /// <exception cref="InvalidOperationException">Lançada se o nome da pessoa estiver vazio ou nulo.</exception>
    public async Task<Person> CreateAsync(Person person)
    {
        if (string.IsNullOrWhiteSpace(person.Name))
            throw new InvalidOperationException("O nome é obrigatório.");
        
        await _personRepository.AddAsync(person);
        return person;
    }

    /// <summary>
    /// Busca os detalhes de uma pessoa escpecífica através de seu identificador.
    /// </summary>
    /// <param name="id">O identificador único da pessoa</param>
    /// <returns>A instância da entidade Person encontrada</returns>
    /// <exception cref="KeyNotFoundException">Lançada caso não exista uma pessoa cadastrada com o identificador fornecido.</exception>
    public async Task<Person> GetByIdAsync(int id)
    {
        var person = await _personRepository.GetByIdAsync(id);
        
        if (person == null)
            throw new KeyNotFoundException($"Pessoa com ID {id} não encontrada.");

        return person;
    }

    /// <summary>
    /// Retorna uma lista contendo todas as pessoas cadastradas no sistema.
    /// </summary>
    /// <returns>Uma coleção enumerável de objetos Person</returns>
    public async Task<IEnumerable<Person>> GetAllAsync()
    {
        return await _personRepository.GetAllAsync();
    }

    /// <summary>
    /// Remove uma pessoa do banco de dados.
    /// </summary>
    /// <param name="id">O identificador únido da pessoa a ser removida.</param>
    /// <remarks>
    /// Este método realiza uma busca prévia para garantir que a pessoa existe no banco.
    /// A deleção pode disparar uma remoção em cascata (Cascade Delete) nas transações vinculadas à pessoa.
    /// </remarks>
    /// <exception cref="KeyNotFoundException">Lançada caso o identificador informado não seja encontrado no banco.</exception>
    public async Task DeleteAsync(int id)
    {   
        var person = await GetByIdAsync(id);
        await _personRepository.DeleteAsync(person);
        
    }
}

File: frontend/src/api/personService.ts
import {api} from './axios';
import {type Person} from "../types/models.ts";

/**
 * Serviço responsável pelas operações CRUD da entidade Person.
 * Centraliza as chamadas da API para o endpoint /people.
 */
export const personService = {

    /**
     * Busca todas as pessoas cadastradas.
     * @returns Uma promessa que resolve para um array de objetos Person.
     */
    async getAll(): Promise<Person[]> {
        const response = await api.get<Person[]>('/people');
        return response.data;
    },

    /**
     * Busca uma pessoa específica pelo seu ID.
     * @param id - O identificador único da pessoa.
     * @returns Uma promessa que resolve para o objeto Person encontrado.
     */
    async getById(id: number): Promise<Person> {
        const response = await api.get<Person>(`/people/${id}`);
        return response.data;
    },

    /**
     * Cria uma nova pessoa no banco de dados.
     * @param data - O objeto com os dados da pessoa, omitindo o seu ID.
     * @returns A resposta da requisição POST.
     */
    async create (data: Omit<Person, 'id'>) {
        return await api.post('/people', data);
    },

    /**
     * Remove uma pessoa do banco de dados através de seu ID.
     * @param id - O identificador único da pessoa a ser removida.
     * @returns A resposta da requisição DELETE.
     */
    async delete (id: number) {
        return await api.delete(`/people/${id}`);
    }
}